src/edu/csus/ecs/pc2/core/Constants.java

```diff
@@ -99,6 +99,22 @@ private Constants() {
     public static final String DEFAULT_CLICS_VALIDATOR_COMMAND = "{:validator} {:infile} {:ansfile} {:feedbackdir} ";
     
     
+    /**
+     * Command substitution variable for name of thing to execute (basename of source file)
+     */
+    public static final String CMDSUB_BASENAME_VARNAME = "{:basename}";
+    
+    /**
+     * Command substitution variable for conditional suffix
+     * This can be used for languages (such as Kotlin), that has a different main class name than the basename of
+     * the source.  eg.  MyClass.kt  uses MyClassKt as the main class name.  This is conditional in that if the
+     * trailing string (suffix) isn't already immediately to the left of the operand of this substitution variable, then it gets added.
+     * eg. {:basename}{:ifsuffix=Kt}  if {:basename} is MyClassKt then, the {:ifsuffix=Kt} is a no-op (empty string).  If
+     * {:basename} is MyClass then the {:ifsuffix=Kt} evaluates to "Kt" making the result: MyClassKt.
+     * Use of this is not just restricted to executables or basename; it can be used anywhere in the string.
+     */
+    public static final String CMDSUB_COND_SUFFIX = "{:ifsuffix=}";
```

Comment:
I find the chosen substitution variable name (`ifsuffix`) to be confusing relative to what is actually being done.  The use of "if" in the name makes it sound like it's something like "if (the suffix is equal to xxx) then add xxx" when in fact it's just the opposite:  what it's doing is saying "if (the suffix is NOT equal to xxx) then add xxx".   This was one of the reasons I was so confused about what the new conditional variable was doing -- i.e., what its purpose was.

A more accurate description of the action being performed would be something like `{:ensuresuffix=}`, which is a lot closer to what is actually being done.   This is a bit longer -- but it's no longer than other substitution variables we already have (e.g. `problemshort`, `languagename`), and it's (IMHO) a much better description of what's being done.   I suggest replacing `{:ifsuffix=}` with `{:ensuresuffix=}`.